### PR TITLE
ROC-4726: Mapping defendant response to settlement agreement

### PR DESCRIPTION
**Pull request by karthikkannankainos**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-4726

### Change description ###
Mapping and creating events for defendant response to settlement agreement

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/CCDPartyStatementAssert.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                if (!Objects.equals(offer.getContent(), actual.getOfferContent())) {
+                    failWithMessage("Expected Offer content to be <%s> but was <%s>",
+                        offer.getContent(), actual.getOfferContent());
+                }
+
+            });
+        }
+
+        Optional.ofNullable(actual.getPaymentIntention()).ifPresent(pymtIntention -> {
+            Assertions.assertThat(partyStatement.getOffer().get().getPaymentIntention().get()).isEqualTo(actual.getPaymentIntention());
```
Comment (dharmendrak) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/CCDPartyStatementAssert.java]: will through null pointer if payment intention is null, please follow your approach as in ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/PartyStatementAssert.java

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCJRequest.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package uk.gov.hmcts.cmc.ccd.domain;
+
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class CCJRequest {
```
Comment (sabahirfan) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCJRequest.java]: Please remove this class, this is not required.

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/defendant/statementofmeans/CCDEmploymentStatus.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.hmcts.cmc.ccd.domain.defendant.statementofmeans;
+
+public enum CCDEmploymentStatus {
```
Comment (sabahirfan) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/defendant/statementofmeans/CCDEmploymentStatus.java]: Please remove this enum as well.

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.Objects;
+import java.util.stream.Stream;
+
+public class MapperUtil {
+
+    private MapperUtil(){
+        // Utility class, no instances
+    }
+
+    public static boolean isAllNull(Object... objects) {
```
Comment (sabahirfan) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java]: 👍

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/DefendantAssert.java [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+import java.util.Objects;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.cmc.ccd.assertion.Assertions.assertThat;
+import static uk.gov.hmcts.cmc.ccd.domain.CCDPartyType.COMPANY;
+import static uk.gov.hmcts.cmc.ccd.domain.CCDPartyType.INDIVIDUAL;
+import static uk.gov.hmcts.cmc.ccd.domain.CCDPartyType.ORGANISATION;
+import static uk.gov.hmcts.cmc.ccd.domain.CCDPartyType.SOLE_TRADER;
+
+public class DefendantAssert extends AbstractAssert<DefendantAssert, TheirDetails> {
```
Comment (sabahirfan) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/DefendantAssert.java]: We have `uk.gov.hmcts.cmc.ccd.assertion.TheirDetailsAssert` in master for the same. Please use that.

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/SaveClaimWithCoreCaseDataStoreTest.java [hunk 6]
```diff
@@ -29,6 +30,7 @@
         "feature_toggles.ccd_enabled=true"
     }
 )
+@Ignore
```
Comment (sabahirfan) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/SaveClaimWithCoreCaseDataStoreTest.java]: Please remove this ignore now.

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/support/ResendStaffNotificationsCoreCaseDataTest.java [hunk 7]
```diff
@@ -49,6 +49,7 @@
         "feature_toggles.ccd_enabled=true"
     }
 )
+@Ignore
```
Comment (sabahirfan) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/support/ResendStaffNotificationsCoreCaseDataTest.java]: Please remove this ignore now.

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/RequestMoreTimeForResponseWithCoreCaseDataTest.java [hunk 8]
```diff
@@ -46,6 +47,7 @@
         "feature_toggles.reminderEmails=true"
     }
 )
+@Ignore
```
Comment (sabahirfan) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/RequestMoreTimeForResponseWithCoreCaseDataTest.java]: Please remove this ignore now.

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/SaveClaimWithCoreCaseDataStoreTest.java [hunk 9]
```diff
@@ -28,8 +28,7 @@
         "feature_toggles.ccd_async_enabled=false",
         "feature_toggles.ccd_enabled=true"
     }
-)
-public class SaveClaimWithCoreCaseDataStoreTest extends BaseSaveTest {
+) public class SaveClaimWithCoreCaseDataStoreTest extends BaseSaveTest {
```
Comment (dharmendrak) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/SaveClaimWithCoreCaseDataStoreTest.java]: something not right here?

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCJRequest.java [hunk 10]
```diff
@@ -0,0 +1,9 @@
+package uk.gov.hmcts.cmc.ccd.domain;
```
[the remark this answers is not in the document]
Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCJRequest.java]: this is outdated change. please review the latest

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/defendant/statementofmeans/CCDEmploymentStatus.java [hunk 11]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.hmcts.cmc.ccd.domain.defendant.statementofmeans;
```
[the remark this answers is not in the document]
Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/defendant/statementofmeans/CCDEmploymentStatus.java]: please review latest file. This is old code you are reviewing

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/SaveClaimWithCoreCaseDataStoreTest.java (hunk 9 above)
Comment (karthikkannankainos) [reply src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/SaveClaimWithCoreCaseDataStoreTest.java]: @dharmendrak next thing we need to incorporate in checkstyle. But revoking your approval for this is quite cruel :-)

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/offers/PartyStatementMapperTest.java [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+
+        //when
+        CCDPartyStatement ccdPartyStatement = partyStatementMapper.to(partyStatement);
+
+        //then
+        assertThat(partyStatement).isEqualTo(ccdPartyStatement);
+    }
+
+    @Test
+    public void mapPartyStatementWithoutOfferToCCD() {
```
Comment (sabahirfan) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/offers/PartyStatementMapperTest.java]: As discussed please rename this test to `mapCountersignPartyStatementsToCCD`. And have the sequence of `partyStatements` as 
```
Arrays.asList(
            SamplePartyStatement.counterSignPartyStatement,
            SamplePartyStatement.acceptPartyStatement,
            SamplePartyStatement.offerPartyStatement
        );
```

**Comment by karthikkannankainos** [comment]
Closing this PR as build pipeline had a issue. I squashed and created a separate branch to get it approved. So wont need this PR anymore

**Pull request closed without merging**